src/sqlfluff/dialects/dialect_mysql.py

```diff
@@ -90,3 +97,49 @@ class CreateTableStatementSegment(
             ),
         ],
     )
+
+mysql_dialect.add(
+    DoubleForwardSlashSegment=StringParser(
+        "//", SymbolSegment, name="doubleforwardslash", type="statement_terminator"
+    ),
+    DoubleDollarSignSegment=StringParser(
+        "$$", SymbolSegment, name="doubledollarsign", type="statement_terminator"
+    ),
+)
+
+mysql_dialect.replace(
+    DelimiterSegment=OneOf(Ref("SemicolonSegment"), Ref("TildeSegment")),
+    TildeSegment=StringParser(
+        "~", SymbolSegment, name="tilde", type="statement_terminator"
+    ), 
+)
+
+@mysql_dialect.segment()
+class FunctionDelimiterGrammar(BaseSegment):
```

Comment:
I suggest omitting the `parse_grammar` for now -- a `parse_grammar` is sufficient in most cases. I notice you nested a `Sequence` inside a `Sequence`, which seems unnecessary, and the inner one is marked `optional`. I'm not sure about that. This is my best guess for now:
```
@mysql_dialect.segment()
class FunctionDelimiterGrammar(BaseSegment):
    parse_grammar = Sequence(
        "DELIMITER",
        OneOf(Ref("DelimiterSegment")),
        optional=True,
    )
```

I think the reason this isn't working is that it's not hooked into the rest of the grammar. When you add a new segment type, it doesn't really have any effect until it is mentioned in some other place as a potential option to be used in that place.

Reply:
I think you'll need to modify the MySQL grammar to inherit from / add to `class StatementSegment` from the ANSI grammar. `StatementSegment` lists all the possible kinds of statements.

Reply:
@barrywhart I updated the PR with your suggestions.  I had to add the `match_grammar` because when I took it out it was giving me an error of :
`AttributeError: type object 'DelimiterGrammar' has no attribute 'match_grammar'`
If I add the `match_grammar` back, then it doesn't error, but it also doesn't match and I still get the original error:
`L:   1 | P:   1 |  PRS | Found unparsable section: 'delimiter'`